sse2neon.h

```diff
@@ -9139,34 +9139,51 @@ FORCE_INLINE int _mm_cmpestrz(__m128i a,
-                vceqq_u16(vreinterpretq_u16_m128i(b), vdupq_n_u16(0));         \
-            uint8x8_t res_a = vshrn_n_u16(equal_mask_a, 4);                    \
-            uint8x8_t res_b = vshrn_n_u16(equal_mask_b, 4);                    \
-            uint64_t matches_a = vget_lane_u64(vreinterpret_u64_u8(res_a), 0); \
-            uint64_t matches_b = vget_lane_u64(vreinterpret_u64_u8(res_b), 0); \
-            la = _sse2neon_ctzll(matches_a) >> 3;                              \
-            lb = _sse2neon_ctzll(matches_b) >> 3;                              \
-        } else {                                                               \
-            uint16x8_t equal_mask_a = vreinterpretq_u16_u8(                    \
-                vceqq_u8(vreinterpretq_u8_m128i(a), vdupq_n_u8(0)));           \
-            uint16x8_t equal_mask_b = vreinterpretq_u16_u8(                    \
-                vceqq_u8(vreinterpretq_u8_m128i(b), vdupq_n_u8(0)));           \
-            uint8x8_t res_a = vshrn_n_u16(equal_mask_a, 4);                    \
-            uint8x8_t res_b = vshrn_n_u16(equal_mask_b, 4);                    \
-            uint64_t matches_a = vget_lane_u64(vreinterpret_u64_u8(res_a), 0); \
-            uint64_t matches_b = vget_lane_u64(vreinterpret_u64_u8(res_b), 0); \
-            la = _sse2neon_ctzll(matches_a) >> 2;                              \
-            lb = _sse2neon_ctzll(matches_b) >> 2;                              \
-        }                                                                      \
+#define SSE2NEON_CMPISTRX_GET_STR_LEN(str, len, imm8)                    \
+    do {                                                                 \
+        if (imm8 & 0x01) {                                               \
+            uint16x8_t equal_mask_##str =                                \
+                vceqq_u16(vreinterpretq_u16_m128i(str), vdupq_n_u16(0)); \
+            uint8x8_t res_##str = vshrn_n_u16(equal_mask_##str, 4);      \
+            uint64_t matches_##str =                                     \
+                vget_lane_u64(vreinterpret_u64_u8(res_##str), 0);        \
+            len = _sse2neon_ctzll(matches_##str) >> 3;                   \
+        } else {                                                         \
+            uint16x8_t equal_mask_##str = vreinterpretq_u16_u8(          \
+                vceqq_u8(vreinterpretq_u8_m128i(str), vdupq_n_u8(0)));   \
+            uint8x8_t res_##str = vshrn_n_u16(equal_mask_##str, 4);      \
+            uint64_t matches_##str =                                     \
+                vget_lane_u64(vreinterpret_u64_u8(res_##str), 0);        \
+            len = _sse2neon_ctzll(matches_##str) >> 2;                   \
+        }                                                                \
     } while (0)
 
+#define SSE2NEON_CMPISTRX_GET_LEN(a, b, la, lb)     \
```

Comment:
It is confusing to have `SSE2NEON_CMPISTRX_GET_LEN` and `SSE2NEON_CMPISTRX_GET_STR_LEN` since they look similar. Can you clarify?

Reply:
`SSE2NEON_CMPISTRX_GET_LEN` is used to get the length of string `a` and `b`.
While `SSE2NEON_CMPISTRX_GET_STR_LEN` is used to get the length of one string (`_mm_cmpistrs` and `_mm_cmpistrz` just need to get the length of string `a` and `b`, respectively).

Reply:
The name `SSE2NEON_CMPISTRX_GET_STR_LEN` contains two `STR` identifiers. It looks a bit weird. We can rename `SSE2NEON_CMPISTRX_GET_LEN` to `SSE2NEON_CMPISTRX_LEN_PAIR`, which retrieves the length of two strings.

Reply:
Similarly, `SSE2NEON_CMPISTRX_LENGTH` looks slightly understandable.

Reply:
Sounds great. I will make another commit to include this suggestion.

Reply:
Done.